lightning/src/ln/channel.rs

```diff
@@ -1672,7 +1672,29 @@ impl<SP: Deref> Channel<SP> where
 				res
 			},
 			ChannelPhase::Funded(mut funded_channel) => {
-				let res = funded_channel.commitment_signed(msg, logger).map(|monitor_update_opt| (None, monitor_update_opt));
+				#[cfg(splicing)]
+				let session_received_commitment_signed = funded_channel
+					.interactive_tx_signing_session
+					.as_ref()
+					.map(|session| session.has_received_commitment_signed())
+					.unwrap_or(false);
+				#[cfg(splicing)]
+				let res = if funded_channel.pending_splice.is_some() && !session_received_commitment_signed {
+					// TODO(splicing): Currently doesn't build until we have the negotiated
+					// `FundingScope` constructed and available for use (#3736).
+					let pending_splice = funded_channel.pending_splice.as_ref().unwrap();
+					funded_channel
+						.splice_initial_commitment_signed(&pending_splice.funding, msg, logger)
```

Comment:
IIRC, #3736 uses an `Option` for the current `FundingScope` being negotiated. So we'll need to account for that. Could also add the field in this PR with it set to `None` so that this can build.